tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -162,7 +200,7 @@ def resources(self, query=None):
                 return []
             raise
         self._cache.save(key, resources)
-        return self.filter_resources(resources)
+        return resources if skip_filtering else self.filter_resources(resources)
```

Comment:
this cache stuff shouldn't be in this leaf method but in resources method, ditto for the filtering.